python_modules/dagster/dagster_tests/core_tests/test_asset_events.py

```diff
@@ -87,3 +88,22 @@ def my_asset():
 
         assert instance.run_ids_for_asset_key(AssetKey("my_asset_name")) == [run_id]
         assert instance.run_ids_for_asset_key(AssetKey("my_other_asset")) == [run_id]
+
+
+def test_asset_partition_materialization_planned_events():
+    @asset(partitions_def=StaticPartitionsDefinition(["a", "b"]))
+    def my_asset():
+        pass
+
+    assets_job = build_assets_job("assets_job", [my_asset])
+
+    with instance_for_test() as instance:
+        result = assets_job.execute_in_process(instance=instance, partition_key="a")
```

Comment:
IMO preferable to use `materialize`, but not a big deal

Reply:
swapped